### PR TITLE
Long alias support for aggregations

### PR DESCRIPTION
**Pull request by MarkDuckworth**
Long alias support for aggregations. PR to feature branch.

### CODE REVIEW

File: Firestore/core/src/remote/remote_objc_bridge.cc [hunk 1]
```diff
@@ -294,57 +299,75 @@ DatastoreSerializer::EncodeAggregateQueryRequest(
   }
```
Comment (cherylEnkidu) [inline Firestore/core/src/remote/remote_objc_bridge.cc:299]: You can use move here
```
uniqueAggregates.insert(std::move(pair));
```

File: Firestore/core/src/remote/remote_objc_bridge.cc [hunk 2]
```diff
@@ -294,57 +299,75 @@ DatastoreSerializer::EncodeAggregateQueryRequest(
...
   for (const auto& aggregatePair : uniqueAggregates) {
-    result->query_type.structured_aggregation_query.aggregations[i].alias =
-        nanopb::MakeBytesArray(aggregatePair.first);
+    // Map all client-side aliases to a unique short-form
+    // alias. This avoids issues with client-side aliases that
+    // exceed the 1500-byte string size limit.
+    std::string clientAlias = aggregatePair.first;
+    std::string serverAlias = absl::StrFormat("aggregation_%d", aggregationNum);
+    auto pair = std::pair<std::string, std::string>(serverAlias, clientAlias);
+    aliasMap.insert(pair);
```
Comment (cherylEnkidu) [inline Firestore/core/src/remote/remote_objc_bridge.cc:314]: using move here

**Review by cherylEnkidu — CHANGES_REQUESTED** [review]
It looks really good! just a little bit language improvement

File: Firestore/core/src/remote/datastore.cc [hunk 3]
```diff
@@ -291,13 +293,14 @@ void Datastore::RunAggregateQueryWithCredentials(
 
   call->Start(
       // TODO(c++14): move into lambda.
```
Comment (cherylEnkidu) [inline Firestore/core/src/remote/datastore.cc:295]: can you remove the todo here?

**Review by cherylEnkidu — APPROVED** [review]
thank you!

File: Firestore/core/src/remote/datastore.cc (hunk 3 above)
Comment (MarkDuckworth) [reply Firestore/core/src/remote/datastore.cc]: Fixed. Thank you.

**Pull request merged**